src/openstudio_lib/AnalyticsHelper.cpp

```diff
@@ -0,0 +1,79 @@
+*  may not use the "OpenStudio" trademark, "OS", "os", or any other confusingly similar designation without specific prior
+*  written permission from Alliance for Sustainable Energy, LLC.
+*
+*  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDER(S) AND ANY CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES,
+*  INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+*  DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER(S), ANY CONTRIBUTORS, THE UNITED STATES GOVERNMENT, OR THE UNITED
+*  STATES DEPARTMENT OF ENERGY, NOR ANY OF THEIR EMPLOYEES, BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL,
+*  EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF
+*  USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+*  STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF
+*  ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+***********************************************************************************************************************/
+
+#include "AnalyticsHelper.hpp"
+
+#include "../model_editor/Utilities.hpp"
+#include "../utilities/OpenStudioApplicationPathHelpers.hpp"
+
+#include <QByteArray>
+#include <QNetworkAccessManager>
+#include <QNetworkRequest>
+#include <QString>
+#include <QUrlQuery>
+
+namespace openstudio {
+
+AnalyticsHelper::AnalyticsHelper(QObject* parent) : QObject(parent), m_networkAccessManager(new QNetworkAccessManager(this)) {
+  m_networkAccessManager->setAutoDeleteReplies(true);
+}
+
+AnalyticsHelper::~AnalyticsHelper() {
+  // cancel all futures
+  delete m_networkAccessManager;
+}
+
+void AnalyticsHelper::sendAnalytics(const QString& analyticsId, int verticalTabIndex) {
+
+  QNetworkRequest request(QString::fromUtf8("https://www.google-analytics.com/collect"));
+  request.setRawHeader("Content-Type", "application/x-www-form-urlencoded");
+
+  QString title = QString("Tab-%1").arg(verticalTabIndex);
+  QString path = QString("/OSApp/Tab-%1").arg(verticalTabIndex);
+  QString version = toQString(openstudio::openStudioApplicationVersionWithPrerelease());
+
+  QUrlQuery query;
+  // https://developers.google.com/analytics/devguides/collection/protocol/v1/parameters
+  query.addQueryItem("v", "1");
+  query.addQueryItem("tid", "UA-172465420-1");
+  query.addQueryItem("cid", analyticsId);
+  query.addQueryItem("t", "pageview");
+  query.addQueryItem("dh", "app.openstudiocoalition.org");
+  query.addQueryItem("dp", path);
+  query.addQueryItem("dt", title);
+  query.addQueryItem("cd", title);
+  query.addQueryItem("an", "OSApp");
+  query.addQueryItem("av", version);
+  query.addQueryItem("z", QString::number(std::time(0)));
+
+  QByteArray body = query.query().toUtf8();
+  m_networkAccessManager->post(request, body);
+}
```

Comment:
Should we capture subtabs? (eg: Loads on the Space Types tab)

Reply:
Also :+1: I confirmed (by adding an `std::cout << body.toStdString() << '\n'` here) that when I tick the checkbox to not allow analytics that not analytics is actually posted.

Reply:
I think we'll have a hard enough time making sense of the tab data to start with.